arch/arm/include/aarch32/cortex_a_r/exc.h

```diff
@@ -38,6 +38,17 @@ static ALWAYS_INLINE bool arch_is_in_isr(void)
 	return (_kernel.cpus[0].nested != 0U);
 }
 
+#if defined(CONFIG_USERSPACE)
+/*
+ * This function is used by privileged code to determine if the thread
+ * associated with the stack frame is in user mode.
+ */
+static ALWAYS_INLINE bool arch_exception_from_user_mode(const z_arch_esf_t *esf)
+{
+	return ((esf->basic.xpsr & CPSR_M_Msk) == CPSR_M_USR) ? true : false;
```

Comment:
The result of `((esf->basic.xpsr & CPSR_M_Msk) == CPSR_M_USR)` is already boolean. Should remove the unnecessary ternary operator here.